Vic2ToHoI4/Source/V2World/Countries/Country.cpp

```diff
@@ -41,6 +41,32 @@ void Vic2::Country::eatCountry(Country& target, bool debug)
 }
 
 
+void Vic2::Country::mergeStates(const StateDefinitions& stateDefinitions)
+{
+	for (auto state = states.begin(); state != states.end(); ++state)
+	{
+		if (!state->isPartialState())
+		{
+			continue;
+		}
+
+		auto state2 = state;
+		++state2;
+		while (state2 != states.end())
+		{
+			if (state->getStateID() != state2->getStateID())
```

Comment:
Wait. so, partial states have identical state IDs to eachother?

Reply:
The state ID comes from the Vic2 state definitions based on which provinces are in the states. So yes. Which is nice because it removed the need for more complex logic to figure out which states were part of each other.